### PR TITLE
plplot: fixes cmake bug

### PR DESCRIPTION
**Pull request by tschoonj**
according to upstream fix
http://sourceforge.net/p/plplot/plplot/ci/772223c638ecf5dc740c9f3dd7a6883c6d2c83d2/

### CODE REVIEW

**Comment by mwcraig** [comment]
@tschoonj -- are you amenable to a PR on your PR here that also adds an upstream change to the variable plplot uses to check whether it is building (changes `HAVE_CONFIG_H` to `PLPLOT_HAVE_CONFIG_H` to avoid collisions with other packages that use `HAVE_CONFIG_H`)?

File: Library/Formula/plplot.rb [hunk 1]
```diff
@@ -12,11 +12,14 @@ class Plplot < Formula
 
   option 'with-java'
 
+  patch :DATA
```
Comment (mwcraig) [inline Library/Formula/plplot.rb]: I'm not a brew maintainer, so take this with a grain of salt, but I think they want a comment above each patch indicating why it is necessary.

Comment (mikemcquaid) [inline Library/Formula/plplot.rb]: Yup! Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

**Comment by tschoonj** [comment]
@mwcraig what is the link to the patch you are referring to?

@mikemcquaid this patch is already coming from upstream, as mentioned in my initial commit message. Sorry about the missing patch message

**Comment by mwcraig** [comment]
The fixes are extracted from http://sourceforge.net/p/plplot/plplot/ci/467c0b0ef58b1759238d7ec3551e3e3fe8f36c5b/

Feel free to grab the patches from the gist link below if you prefer to include them at the end of the formula.

```ruby
  # Fix from upstream to avoid conflicting definitions of HAVE_CONFIG_H
  # See http://sourceforge.net/p/plplot/bugs/146/
  patch do
    url "https://gist.githubusercontent.com/mwcraig/5059504e65dd0b804bd8/raw/8869cd9b6d188bad9be87c23154bfe0bde054f62/better_fix.diff"
    sha1 "6e9bf8e09d500546cfe872c3f3d92b4ffe39588c"
  end
```

**Comment by tschoonj** [comment]
ok done

**Pull request closed without merging**